edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/util/MetricsDeDuplication.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.Diagnostics.Util
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+
+    public static class MetricsDeDuplication
+    {
+        public static IEnumerable<Metric> RemoveDuplicateMetrics(IEnumerable<Metric> metrics)
```

Comment:
I don't like the name here - You are esentially filtering one enumerable into another enumerable. You could make this an extension method called GetDistinct(). This would allow you to write code like: `var foo = metrics.GetDistinct();`

Reply:
renamed to RemoveDuplicateMetrics and made extension function